File: Blog.Domain/Article.cs
namespace Blog.Domain;

public class Article
{
    public int Id { get; set; }
    
    public User Author { get; set; }
    
    public string Title { get; set; }
    
    public string Type { get; set; }
    
    public ICollection<Comment> Comments { get; set; }
    
    public string Content { get; set; }
    
    public string Image { get; set; }
    
    public Template Template { get; set; }
    
    public DateTime CreatedAt { get; set; }
    
    public DateTime UpdatedAt { get; set; }

    public DateTime DeletedAt { get; set; }

    public void ValidOrFail()
    {
        throw new NotImplementedException();
    }
}

public enum Template
{
    Left,
    Right,
    Center
}

File: client/projects/angular-template/src/app/features/shared/domain/article.ts
import {Comment} from "./comments"

export interface Article{
  id: number
  title: string
  private: boolean
  comments : Comment[]
  content: string
  firstImage : string;
  secondImage : string;
  template: number
  createdAt: string
  updatedAt: Date;
  
  
  isApproved ?: boolean
  isRejected ?: boolean
  
  // One or another
  authorId?: number | null;
  author?: Author | null;
}

export interface Author {
  id: number
  firstName: string
  lastName: string
  username: string
  email: string
}


export interface ArticleListVM {
  entities: Article[]
  loading: boolean
  error : string | null
}


export interface OffensiveWord{
  id : number;
  word : string;
}


export enum ArticleStatus{
  Approved = 'approved',
  Pending = 'pending',
  Rejected = 'rejected'
}
